lhotse/recipes/vcc2018.py

```diff
@@ -0,0 +1,329 @@
+    assert mos_dir.is_dir(), f"No such directory: {mos_dir}"
+    mos_scores = mos_dir / "vcc2018_evaluation_mos.txt"
+    assert mos_scores.is_file()
+    sim_scores = mos_dir / "vcc2018_evaluation_sim.txt"
+    sim_scores.is_file()
+
+    trn_dir = corpus_dir / "scripts"
+    assert trn_dir.is_dir(), f"No such directory: {trn_dir}"
+
+    reference_speech_dir = corpus_dir / "vcc2018_reference"
+    assert reference_speech_dir.is_dir(), f"No such directory: {reference_speech_dir}"
+
+    logging.info(
+        f"Collecting reference target recordings for the VCC2018 challange from {reference_speech_dir}"
+    )
+
+    # TODO
+    # def from_file(p: Path, prefix):
+    #     # Match the format of converted recordings  B00_VCC2TF1_VCC2SF1_30001_HUB.wav
+    #     return Recording.from_file(
+    #         p, recording_id=f"reference_reference_{p.parent.stem}_{p.stem}_reference"
+    #     )
+    #
+    # reference_target_recordings = RecordingSet.from_recordings(
+    #     tqdm(
+    #         map(from_file, reference_speech_dir.rglob("*.wav")),
+    #         desc="Collecting reference recordings",
+    #     )
+    # )
+
+    logging.info(
+        f"Collecting converted recordings submitted to the VCC2018 challange from {converted_speech_dir}"
+    )
+    converted_recordings = RecordingSet.from_dir(
+        converted_speech_dir, "*.wav", num_jobs=num_jobs
+    )
+
+    # # TODO distinguish based on prefix
+    # # recordings = converted_recordings + reference_target_recordings
+    # recordings = reference_target_recordings
+    recordings = converted_recordings
+
+    id2trn = {}
+    for trnp in trn_dir.glob("*.txt"):
+        with open(trnp, "rt") as r:
+            id2trn[trnp.stem] = r.read().strip()
+
+    supervisions = prepare_mos_supervisions(mos_scores, recordings, id2trn)
+    validate_recordings_and_supervisions(recordings, supervisions)
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        recordings.to_jsonl(output_dir / "recordings.jsonl.gz")
+        supervisions.to_jsonl(output_dir / "supervisions.jsonl.gz")
+
+    return {"recordings": recordings, "supervisions": supervisions}
+
+
+def prepare_mos_supervisions(
+    mos_results_path, recordings: RecordingSet, id2trn: Dict[str, str]
+) -> SupervisionSet:
+    # TODO very slow -> make it faster it takes ~8min 170it/s
+    # Use sort & group by instead of O(n*n) select
+
+    mos = load_vcc_results(mos_results_path)[
+        [
+            "test_id",
+            "user_id",
+            "set_id",
+            "set_idx",
+            "system1_id",
+            "src_spk",
+            "tgt_spk",
+            "MOS",
+            "left_audio",
+        ]
+    ]
+    recording_ids = set(mos["left_audio"].tolist())
+    supervisions = []
+    for recording_id_wav in tqdm(recording_ids, desc="Supervision creation"):
+        recording_id = recording_id_wav.rstrip(".wav")
```

Comment:
If you make the `recording_ids` above as a set of `Path` types, then you could use `.stem` here instead.

Reply:
I know… but it is less readible with `Path(recording_id_wav).stem` since I literally need to use both recording_id wit hand without “.wav” for the original data and Lhotse use.

Reply:
`recording_id_wav.name` would give you the file name with the extension.